File: final_dataset_A/generated_code/claude/python/parse_product_catalog_json-claude.py
import json

def parse_catalog(json_file):
    with open(json_file, 'r') as f:
        data = json.load(f)

    products = data['products']
    for p in products:
        print(p['name'], p['price'], p['category'])
    return products

if __name__ == '__main__':
    parse_catalog('catalog.json')

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/ParseProductCatalogJSON-claude.java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class ParseProductCatalogJSON {

    public static List<JsonNode> parseCatalog(String filePath) throws Exception {
        ObjectMapper mapper = new ObjectMapper();
        JsonNode root = mapper.readTree(new File(filePath));
        JsonNode productsNode = root.get("products");

        List<JsonNode> products = new ArrayList<>();
        for (JsonNode product : productsNode) {
            products.add(product);
        }
        return products;
    }

    public static void main(String[] args) throws Exception {
        List<JsonNode> products = parseCatalog("catalog.json");
        for (JsonNode p : products) {
            System.out.println(p.get("name").asText() + " - " + p.get("price").asDouble());
        }
    }
}
